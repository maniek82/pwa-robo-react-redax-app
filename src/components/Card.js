import React from 'react';


const Card = ({ name, email, city, photo, age, phone }) => {
  return (
    <div className='tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5'>
      <img alt='robots' src={`https://robohash.org/${email}?size=100x100`} />
      <div>
        <h4 >Name: {name}</h4>
        <p> {email}</p>
        <p> Tel: {phone}</p>
        <p>Country: {city}</p>
        <p>Age: {age}</p>
        <img style={{width: '220px', height:'auto'}}alt='women' src={`${photo}`} />
      </div>
    </div>
  );
}

export default Card;