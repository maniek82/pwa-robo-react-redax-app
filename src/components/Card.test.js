import {shallow} from 'enzyme'
import React from 'react'
import Card from './Card';



describe('Card props', () => {


it('expect to render Card component',()=> {
    expect(shallow(<Card />).length).toEqual(1)

})

it('It includes img tag',()=> {
    const wrapper = shallow(<Card />)
    expect(wrapper.find('img')).toBeDefined();
})

it('It includes class tc',()=> {
    const wrapper = shallow(<Card />)
    expect(wrapper.find('.tc')).toBeDefined();
})
it('It includes paragraph with text',()=> {

    const wrapper = shallow(<Card />)
    expect(wrapper.contains(<p>Age: </p>)).toBeTruthy();
})
it('It includes paragraph with non existing text',()=> {
    expect.assertions(2)

    const wrapper = shallow(<Card />)
    expect(wrapper.contains(<p>Age: </p>)).toBeTruthy();
    expect(wrapper.contains(<h2>Age: </h2>)).toBeFalsy();
})
})

describe('Card rendering', () => {

    it('should render correctly Card', () => {
      const component = shallow(<Card />);
    
      expect(component).toMatchSnapshot();
    });

    it('should render banner text correctly with given strings', () => {
        const strings = ['one', 'two'];
        const component = shallow(<Card list={strings} />);
        expect(component).toMatchSnapshot();
      });
  });