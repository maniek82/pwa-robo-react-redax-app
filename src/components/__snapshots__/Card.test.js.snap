// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card rendering should render banner text correctly with given strings 1`] = `
<div
  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?size=100x100"
  />
  <div>
    <h4>
      Name: 
    </h4>
    <p>
       
    </p>
    <p>
       Tel: 
    </p>
    <p>
      Country: 
    </p>
    <p>
      Age: 
    </p>
    <img
      alt="women"
      src="undefined"
      style={
        Object {
          "height": "auto",
          "width": "220px",
        }
      }
    />
  </div>
</div>
`;

exports[`Card rendering should render correctly Card 1`] = `
<div
  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?size=100x100"
  />
  <div>
    <h4>
      Name: 
    </h4>
    <p>
       
    </p>
    <p>
       Tel: 
    </p>
    <p>
      Country: 
    </p>
    <p>
      Age: 
    </p>
    <img
      alt="women"
      src="undefined"
      style={
        Object {
          "height": "auto",
          "width": "220px",
        }
      }
    />
  </div>
</div>
`;
